src/_guides/language/sound-problems.md

```diff
@@ -122,6 +119,71 @@ Otherwise, use `dynamic` in situations where you cannot use a single type:
 var width = canvasOrImg.width;
 {% endprettify %}
 
+#### Example 2: Omitted type parameters default to their type bounds
+
+Consider the following **generic class** with a **bounded type parameter** that extends
+`Iterable`:
+
+<?code-excerpt "strong/lib/bounded/my_collection.dart"?>
+{% prettify dart %}
+class C<T extends Iterable> {
+  final T collection;
+  C(this.collection);
+}
+{% endprettify %}
+
+The following code creates a new instance of this class (omitting the type
+argument) and accesses its `collection` member:
+
+{:.fails-sa}
+<?code-excerpt "strong/lib/bounded/instantiate_to_bound.dart (undefined_method)" replace="/c\..*;/[!$&!]/g"?>
+{% prettify dart %}
+var c = new C(Iterable.empty()).collection;
+[!c.add(2);!]
+{% endprettify %}
+
+{:.console-output}
+<?code-excerpt "strong/analyzer-2-results.txt" retain="/add.*isn't defined for the class/"?>
+```nocode
+error • The method 'add' isn't defined for the class 'Iterable' at lib/bounded/instantiate_to_bound.dart:7:5 • undefined_method
+```
+
+While the [List][] type has an `add()` method, [Iterable][] does not.
+
+#### Fix: Specify type arguments or fix downstream errors
+
+In Dart 1.x, when a generic class is instantiated without explicit type
+arguments, `dynamic` is assumed. That is why, in the code excerpt above, `c` is
+of type `dynamic` and no error is reported for `c.add()`.
+
+In Dart 2, when a generic class is instantiated without explicit type arguments,
+each type parameter defaults to its type bound (`Iterable` in this example) if
+one is explicitly given, or `dynamic` otherwise.
+
+You'll need to approach fixing such errors on a case-by-case basis. It helps to
+have a good understanding of the original design intent.
+
+Explicitly passing type arguments is one way to help identify type errors. For
+example, by specifying `List` as a type argument, the analyzer let's you know
+that the constructor argument has the wrong type and needs to be changed:
+
+{:.passes-sa}
+<?code-excerpt "strong/test/strong_test.dart (add-type-arg)" replace="/.List.|\[\]/[!$&!]/g"?>
+{% prettify dart %}
+var c = new C[!<List>!]([![]!]).collection;
+c.add(2);
+{% endprettify %}
+
+If you actually meant `collection` to be an `Iterable`, then subsequent uses of
```

Comment:
I'm confused about this. Could you clarify it?

Reply:
Actually, I think that I'll just drop this (short) paragraph and its code example.